Spec/Auth.swift

```diff
@@ -205,6 +205,272 @@ class Auth : QuickSpec {
+                            let realtime = ARTRealtime(options: options)
+                            defer { realtime.dispose(); realtime.close() }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                realtime.connection.once(.Disconnected) { stateChange in
+                                    guard let stateChange = stateChange else {
+                                        fail("ConnectionStateChange is nil"); done(); return
+                                    }
+                                    guard let errorInfo = stateChange.reason else {
+                                        fail("ErrorInfo is nil"); done(); return
+                                    }
+                                    expect(errorInfo.code) == 80019
+                                    done()
+                                }
+                            }
+
+                            guard let errorInfo = realtime.connection.errorReason else {
+                                fail("ErrorInfo is empty"); return
+                            }
+                            expect(errorInfo.code) == 80019
+                        }
+
+                        // RSA4c3
+                        it("if the connection is CONNECTED, then the connection should remain CONNECTED") {
+                            let token = getTestToken(ttl: 2.0)
+                            let options = AblyTests.clientOptions()
+                            options.authUrl = NSURL(string: "http://echo.ably.io")!
+                            options.authParams = [NSURLQueryItem]()
+                            options.authParams?.append(NSURLQueryItem(name: "type", value: "text"))
+                            options.authParams?.append(NSURLQueryItem(name: "body", value: token))
+
+                            let realtime = ARTRealtime(options: options)
+                            defer { realtime.dispose(); realtime.close() }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                realtime.connection.once(.Connected) { stateChange in
+                                    expect(stateChange?.reason).to(beNil())
+                                    done()
+                                }
+                            }
+
+                            // Token should renew and fail
+                            realtime.options.authParams = [NSURLQueryItem]()
+
+                            expect(realtime.connection.errorReason).toEventuallyNot(beNil(), timeout: testTimeout)
+                            guard let errorInfo = realtime.connection.errorReason else {
+                                fail("ErrorInfo is empty"); return
+                            }
+                            expect(errorInfo.code) == 80019
+
+                            expect(realtime.connection.state).to(equal(ARTRealtimeConnectionState.Connected))
```

Comment:
I don't think this test can pass until http://docs.ably.io/client-lib-development-guide/features/#RTC8a is implemented. Currently the token would expire, Ably would disconnect the connection, and then the connection would become disconnected.  Instead what you want is for the connection be made with a token that does not expire, you then inject an `AUTH ProtocolMessage` so the client thinks the server has requested it to authorise, it authorises fails and then just leaves the connection in the CONNECTED state waiting for the server to disconnect it eventually.